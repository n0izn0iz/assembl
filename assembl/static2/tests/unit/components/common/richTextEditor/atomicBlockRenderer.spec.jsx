import React from 'react';
import renderer from 'react-test-renderer';

import AtomicBlockRenderer from '../../../../../js/app/components/common/richTextEditor/atomicBlockRenderer';

function Entity(data, type) {
  this.data = data;
  this.type = type;
  this.getData = function () {
    return this.data;
  };
  this.getType = function () {
    return this.type;
  };
}

describe('AtomicBlockRenderer component', () => {
  it('should render an image block', () => {
    const data = {
      title: 'Foobar',
      externalUrl: 'http://www.example.com/foobar.jpg',
      mimeType: 'image'
    };
    const fakeEntity = new Entity(data, 'document');
    const fakeContentState = {
      getEntity: (key) => {
        if (key === '0') {
          return fakeEntity;
        }
        return null;
      }
    };
    const fakeBlock = {
      getEntityAt: (offset) => {
        return offset.toString(); // fake entity key
      }
    };

    const props = {
      block: fakeBlock,
      contentState: fakeContentState
    };

    const component = renderer.create(<AtomicBlockRenderer {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});