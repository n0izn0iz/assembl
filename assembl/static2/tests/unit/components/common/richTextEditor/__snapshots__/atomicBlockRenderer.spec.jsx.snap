// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtomicBlockRenderer component should render an image block 1`] = `
<figure>
  <img
    alt=""
    src="http://www.example.com/foobar.jpg"
    title="Foobar"
    width="60%"
  />
</figure>
`;
